app/models/atmosphere/virtual_machine_template.rb

```diff
@@ -144,6 +144,12 @@ def perform_delete_in_cloud
       logger.warn("VMT with #{id_at_site} does not exist - continuing")
     end
 
+    # Returns the hourly cost for this template assuming a given VM flavor
+    def get_hourly_cost_for(flavor)
+      incarnation = flavor.flavor_os_families.find_by(os_family: appliance_type.os_family)
+      incarnation && incarnation.hourly_cost
+    end
```

Comment:
It's fine with me the way it is. If to be changed, I would vote for neither solution. In fact, I would simply write it as "incarnation.hourly_cost" and that's it. Exporting "nil exception" higher in the stack does nothing except making the bug search time longer. Raising exception - this is only good as long as you have a serious plan what to do with it. If not (i.e. no appropriate code somewhere higher up, to catch it and act accordingly) this is just changing "nil exception" into "my exception". So -> not worth it.

Now I know this "a && a.call" construct is prevalent these days in open source projects. Perhaps there are places this makes more sense (i.e. when the method is supposed to return an AR object, like a scope or a specific find method, and then receiving nil is to be expected and properly dealt with on the "other side"). In most of the cases, though, in my opinion, it just makes the code more cryptic, longer, and does not "save the day" at all when something wrong happens.

So, @nowakowski, just leave it as it is :).